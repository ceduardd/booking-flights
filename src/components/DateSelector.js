import DatePicker from 'react-date-picker';
import { useSelector } from 'react-redux';
import DropdownIcon from './DropdownIcon';

const DateSelector = ({
  label,
  value,
  changeHandler,
  disabled,
  errorMessage,
}) => {
  const { errorDate } = useSelector(state => state.ui);

  return (
    <label className="date-selector" htmlFor="from">
      {label}:
      <div style={{ marginTop: '5px' }} className="form-group">
        <DatePicker
          value={value}
          onChange={changeHandler}
          clearIcon={null}
          calendarIcon={<DropdownIcon />}
          disabled={disabled}
        />
        <i className="far fa-calendar"></i>
      </div>
      {errorDate && errorDate === errorMessage && (
        <div className="error-label">
          <i className="fas fa-exclamation-circle"></i> {errorMessage}
        </div>
      )}
    </label>
  );
};

export default DateSelector;
