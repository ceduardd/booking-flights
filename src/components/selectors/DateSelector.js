import DatePicker from 'react-date-picker';
import { useSelector } from 'react-redux';
import DropdownIcon from '../icons/DropdownIcon';
import ErrorLabel from '../ErrorLabel';

const DateSelector = ({
  label,
  value,
  changeHandler,
  disabled,
  errorMessage,
}) => {
  const { errorDate } = useSelector(state => state.ui);

  return (
    <label className="date-selector" htmlFor="from">
      <span>{label}:</span>
      <div style={{ marginTop: '5px' }} className="form-group">
        <DatePicker
          value={value}
          onChange={changeHandler}
          clearIcon={null}
          calendarIcon={<DropdownIcon />}
          disabled={disabled}
        />
        <i className="far fa-calendar"></i>
      </div>
      {errorDate && errorDate === errorMessage && (
        <ErrorLabel>{errorMessage}</ErrorLabel>
      )}
    </label>
  );
};

export default DateSelector;
