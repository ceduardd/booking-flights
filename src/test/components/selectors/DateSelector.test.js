import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import 'jest-canvas-mock';

import DateSelector from '../../../components/selectors/DateSelector';

const mockStore = configureStore();

const initialState = {
  ui: {},
};

let store = mockStore(initialState);

store.dispatch = jest.fn();

const changeHandler = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <DateSelector changeHandler={changeHandler} />
  </Provider>
);

describe('<DateSelector />', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    jest.clearAllMocks();
  });

  test('should be match with snapshot', () => {
    // Should show a console.error() because <DatePiker />
    expect(wrapper).toMatchSnapshot();
  });

  test('should show error message', () => {
    const errorMessage = 'Something error';

    const state = {
      ui: {
        errorDate: errorMessage,
      },
    };

    const store = mockStore(state);

    const wrapper = mount(
      <Provider store={store}>
        <DateSelector errorMessage={errorMessage} />
      </Provider>
    );

    expect(wrapper.find('ErrorLabel').exists()).toBe(true);
  });

  test('should call changeHandler', () => {
    const dateValue = new Date();

    const input = wrapper.find('DatePicker');

    // console.log(input.html());

    input.prop('onChange')(dateValue);

    expect(changeHandler).toHaveBeenCalled();
  });
});
