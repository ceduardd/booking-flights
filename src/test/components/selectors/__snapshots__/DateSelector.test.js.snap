// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateSelector /> should be match with snapshot 1`] = `
<label
  className="date-selector"
  htmlFor="from"
>
  <span>
    :
  </span>
  <div
    className="form-group"
    style={
      Object {
        "marginTop": "5px",
      }
    }
  >
    <div
      className="react-date-picker react-date-picker--closed react-date-picker--enabled"
      onFocus={[Function]}
    >
      <div
        className="react-date-picker__wrapper"
      >
        <div
          className="react-date-picker__inputGroup"
          onClick={[Function]}
        >
          <input
            max="275760-09-12"
            min="0000-12-31"
            name="date"
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "position": "absolute",
                "visibility": "hidden",
                "zIndex": "-999",
              }
            }
            type="date"
            value=""
          />
          <input
            autoComplete="off"
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
            data-input="true"
            inputMode="numeric"
            key="input"
            max={12}
            min={1}
            name="month"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="--"
            required={null}
            type="number"
            value=""
          />
          <span
            className="react-date-picker__inputGroup__divider"
          >
            /
          </span>
          <input
            autoComplete="off"
            autoFocus={false}
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
            data-input="true"
            inputMode="numeric"
            key="input"
            max={31}
            min={1}
            name="day"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="--"
            required={null}
            type="number"
            value=""
          />
          <span
            className="react-date-picker__inputGroup__divider"
          >
            /
          </span>
          <input
            autoComplete="off"
            autoFocus={false}
            className="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
            data-input="true"
            inputMode="numeric"
            key="input"
            max={275760}
            min={1}
            name="year"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="----"
            required={null}
            step={1}
            type="number"
            value=""
          />
        </div>
        <button
          className="react-date-picker__calendar-button react-date-picker__button"
          onClick={[Function]}
          onFocus={[Function]}
          type="button"
        >
          <i
            className="fas fa-chevron-down"
          />
        </button>
      </div>
    </div>
    <i
      className="far fa-calendar"
    />
  </div>
</label>
`;
